NETOBSERV-1911 CLI metrics

jpinsonneau opened the pull request:
## Description

Add metrics capture to CLI. Use:
```
$ ./build/oc-netobserv metrics --enable_pktdrop="true" --enable_dns="true" --enable_rtt="true"
```
to run a capture and generate a dashboard.

![screencapture-console-openshift-console-apps-rosa-km3ao-kvjvq-mao-ktcw-p3-openshiftapps-monitoring-dashboards-netobserv-cli-2024-10-18-13_41_30](https://github.com/user-attachments/assets/aef35025-83d8-4550-a2f7-b12b8037de99)

## Dependencies

Based on https://github.com/netobserv/network-observability-cli/pull/103

## Checklist

If you are not familiar with our processes or don't know what to answer in the list below, let us know in a comment: the maintainers will take care of that.

* [X] Is this PR backed with a JIRA ticket? If so, make sure it is written as a title prefix _(in general, PRs affecting the NetObserv/Network Observability product should be backed with a JIRA ticket - especially if they bring user facing changes)._
* [ ] Does this PR require product documentation?
  * [ ] If so, make sure the JIRA epic is labelled with "documentation" and provides a description relevant for doc writers, such as use cases or scenarios. Any required step to activate or configure the feature should be documented there, such as new CRD knobs.
* [ ] Does this PR require a product release notes entry?
  * [ ] If so, fill in "Release Note Text" in the JIRA.
* [ ] Is there anything else the QE team should know before testing? E.g: configuration changes, environment setup, etc.
  * [ ] If so, make sure it is described in the JIRA ticket.
* QE requirements (check 1 from the list):
  * [X] Standard QE validation, with pre-merge tests unless stated otherwise.
  * [ ] Regression tests only (e.g. refactoring with no user-facing change).
  * [ ] No QE (e.g. trivial change with high reviewer's confidence, or per agreement with the QE team).

jotak reviewed (commented):
A couple of comments, no big blocker (just worrying in case things go wrong when stopping capture, trying to imagine some worst case scenario, what would happen)

jpinsonneau commented on Dockerfile:
```diff
@@ -24,6 +24,12 @@ COPY .mk/ .mk/
+RUN set -x; \
+    OC_TAR_URL="https://mirror.openshift.com/pub/openshift-v4/$(uname -m)/clients/ocp/latest/openshift-client-linux.tar.gz" && \
+    curl -L -q -o /tmp/oc.tar.gz "$OC_TAR_URL" && \
+    tar -C /tmp -xvf /tmp/oc.tar.gz oc kubectl
```
CLI runs on both so having both in the Dockerfile make sense to me.

However, the Dockerfile only generates oc-commands for now so I can either keep only oc here or embedd kubectl-commands too. 

What do you prefer ?

jotak replied:
no worries, nothing important if we keep both of them

jpinsonneau commented on README.md:
```diff
@@ -107,24 +109,34 @@ or `dbeaver`:
 To stop capturing press Ctrl-C.
 
-This will write pcap into a single file located in `./output/pcap/<CAPTURE_DATE_TIME>.pcap` that can be opened with Wireshark for example:
+This will write pcapng into a single file located in `./output/pcap/<CAPTURE_DATE_TIME>.pcapng` that can be opened with Wireshark for example:
```
I don't think so. However, most of the tools manages pcapng (wireshark, tshark etc)

jpinsonneau commented on cmd/root.go:
```diff
@@ -115,3 +128,34 @@ func showKernelVersion() {
+			resetTerminal()
+			out, err := exec.Command("/oc-netobserv", "stop").Output()
+			if err != nil {
+				log.Fatal(err)
```
I can put a loop and retry every X seconds indeed.

The only alternative way I have in mind would be to create a Job outside of this collector Pod to invoke the oc delete daemonset command... However, the job will need to get from somewhere the current namespace + the capture ended trigger so I'm not sure it's better somehow.

jpinsonneau replied:
@jotak any thoughts on that ? Else I will merge this PR as is and we can improve later

jotak commented on Dockerfile:
```diff
@@ -24,6 +24,12 @@ COPY .mk/ .mk/
 # Build collector
 RUN GOARCH=$TARGETARCH make compile
 
+# Install oc to allow collector to run commands
```
hmm .. am I missing something? I don't see it in Dockerfile.downstream :)

memodi commented:
/ok-to-test

jpinsonneau commented:
> @jpinsonneau I tested this, I think labelling the CLI namespace to have openshift.io/cluster-monitoring: "true" is missing, I could only get metrics to be scraped after I added that label. Not sure if that's going to be only downstream thing.

That's right @memodi. I have rebased and forced the label in the namespace creation. Thanks !

jpinsonneau commented:
![image](https://github.com/user-attachments/assets/2b9398f3-9456-4b37-bfab-fe87b6bc3837)

memodi commented:
/label qe-approved

jotak commented:
/lgtm
